pkgs/build-support/singularity-tools/default.nix

```diff
@@ -105,14 +122,14 @@ rec {
 
             # Create runScript and link shell
             if [ ! -e bin/sh ]; then
-              ln -s ${runtimeShell} bin/sh
+              ln -s ${lib.getExe bashInteractive} bin/sh
```

Comment:
Why is `runtimeShell` wrong and `bashInteractive` right? I mean, seems to make sense, but I fail to spell it out

Reply:
`runtimeShell` is provided by `runtimeShellPackage` (i.e., `bash`). Current implementation links `runtimeShell` to container's `/bin/sh`, which is then executed by `runscript`.

When calling `apptainer run` or `singularity run`, it executes `runscript` and starts a (supposedly) interactive session. Due to the lack of line-editing functionality of the non-interacive `bash`, control characters, including arrow keys, are not handled correctly.

This change solves this problem by linking to `/bin/sh` `bashInteractive` instead, the way NixOS does by default.

You can see the difference by running

```sh
apptainer run "$(nix-build -A apptainer.tests.image-hello-cowsay)"
```

Before and after this change.